### PR TITLE
Refactor Batch Commands

### PR DESCRIPTION
**Pull request by BoonKhoonSim**
- GenerateCreateKeyPayload() and GenerateLocatePayload() will only generate payload
- BatchCmdAddItem() will add in the input command into a list
- BatchCmdGenerateMessage() will generate the final batch command message to be send to server
- SendRequestMessage() will handle both batch and non-batch messages

### CODE REVIEW

File: src/kmipapi/clientops.go [hunk 1]
```diff
@@ -158,11 +165,25 @@ type QueryRequest struct {
...
+	AttestationCapability   bool                      // Required: No
+	BatchUndoCapability     bool                      // Required: No
+	BatchContinueCapability bool                      // Required: No
+	UnwrapMode              kmip14.UnwrapMode         // Required: No
+	DestroyAction           kmip14.DestroyAction      // Required: No
+	ShreddingAlgorithm      kmip14.ShreddingAlgorithm // Required: No
+	RNGMode                 kmip14.RNGMode            // Required: No
+	QuantumSafeCapability   bool                      // Required: No
+}
+
```
Comment (jskazinski) [inline src/kmipapi/clientops.go]: This type is already defined in op_query.go and should be deleted here.

File: src/kmipapi/clientops.go [hunk 2]
```diff
@@ -158,11 +165,25 @@ type QueryRequest struct {
...
 type QueryResponse struct {
 	// Contains all attributes of the query response operation that are relevant to the caller.
 	Operation            []kmip14.Operation  `json:"Operation,omitempty"`
 	ObjectType           []kmip14.ObjectType `json:"Object Type,omitempty"`
 	VendorIdentification string              `json:"Vendor Identification,omitempty"`
+	CapabilityInformation CapabilityInformation
```
Comment (jskazinski) [inline src/kmipapi/clientops.go]:
```suggestion
	CapabilityInformation kmip.CapabilityInformation
```

File: src/kmipapi/kmip14.go [hunk 3]
```diff
@@ -53,17 +53,18 @@ func (kmips *kmip14service) Query(ctx context.Context, settings *ConfigurationSe
...
 
 	// Extract the QueryResponsePayload type of message
 	var respPayload struct {
-		Operation            []kmip14.Operation
-		ObjectType           []kmip14.ObjectType
-		VendorIdentification string
+		Operation             []kmip14.Operation
+		ObjectType            []kmip14.ObjectType
+		VendorIdentification  string
+		CapabilityInformation CapabilityInformation
```
Comment (jskazinski) [inline src/kmipapi/kmip14.go]:
```suggestion
		CapabilityInformation kmip.CapabilityInformation
```

File: src/kmipapi/send.go [hunk 4]
```diff
@@ -82,7 +115,7 @@ func SendRequestMessage(ctx context.Context, settings *ConfigurationSettings, op
 		}
 
 		// TODO: Need to handle more than more batch item in the future.
-
+		i := respMsg.ResponseHeader.BatchCount - 1
```
Comment (jskazinski) [inline src/kmipapi/send.go]: Need to adjust the TODO comment and also add a comment why the code is just grabbing the last element of the response.

**Comment by jskazinski** [comment]
Please run `gofumpt -w -l .`, as is done in the Makefile, so that the files remain formatting the same as the upstream formatting.

File: src/kmipapi/clientops.go (hunk 1 above)
Comment (BoonKhoonSim) [reply src/kmipapi/clientops.go]: This struct has been removed in newer code.

File: src/kmipapi/clientops.go (hunk 2 above)
Comment (BoonKhoonSim) [reply src/kmipapi/clientops.go]: CapabilityInformation CapabilityInformation has been replaced with CapabilityInformation interface{}.
This is to cater to different CapabilityInformation struct declaration for kmip and kmip20.

File: src/kmipapi/send.go (hunk 4 above)
Comment (BoonKhoonSim) [reply src/kmipapi/send.go]: New code to check thru the returned status of all the batch commands.

**Review by jskazinski — APPROVED** [review]
Looks good to me

**Pull request merged**